### ISSUE TITLE
The Confluence of Geometry and Learning  The Berkeley Artificial Intelligence Research Blog

### ISSUE THREAD

**Issue by guevara** [guevara/read-it-later#1209]
The Confluence of Geometry and Learning – The Berkeley Artificial Intelligence Research Blog<br>
<br>
https://ift.tt/2tED2ET<br>
<br>
Daniel Seita<br>
<br>

<div>
<div>
  <h1>The Confluence of Geometry and Learning</h1>
  <p>
  
  Shubham Tulsiani and Tinghui Zhou   
  
  
  Jul 11, 2017
  
  </p>
</div>

<article>
  <p>Given only a single 2D image, humans are able to effortlessly infer the rich 3D structure of the underlying scene. Since inferring 3D from 2D is an ambiguous task by itself (see e.g. the left figure below), we must rely on learning from our past visual experiences. These visual experiences solely consist of 2D projections (as received on the retina) of the 3D world. Therefore, the learning signal for our 3D perception capability likely comes from making consistent connections among different perspectives of the world that only capture <em>partial</em> evidence of the 3D reality. We present methods for building 3D prediction systems that can learn in a similar manner.</p>

<table>
  <thead>
    <tr>
      <th><img src="http://bair.berkeley.edu/static/blog/confluence/sinha.png" alt="sinha"></th>
      <th><img src="http://bair.berkeley.edu/static/blog/confluence/dpe8C7u.png" alt="MVS"></th>
      <th><img src="http://bair.berkeley.edu/static/blog/confluence/sample_result.png" alt="sampleres"></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><sub><em>An image could be the projection of infinitely many 3D structures (figure from <a href="http://ieeexplore.ieee.org/document/378224/">Sinha &amp; Adelson</a>).</em></sub></td>
      <td><sub><em>Our visual experiences solely comprise of 2D projections of the 3D world.</em></sub></td>
      <td><sub><em>Our approach can learn from 2D projections and predict shape (top) or depth (bottom) from a single image.</em></sub></td>
    </tr>
  </tbody>
</table>

<p>Building computational models for single image 3D inference is a long-standing problem in computer vision. Early attempts, such as the <a href="https://dspace.mit.edu/handle/1721.1/11589">Blocks World</a> or <a href="http://web.mit.edu/cocosci/Papers/Barrow-Tenenbaum81.pdf">3D surface from line drawings</a>, leveraged explicit reasoning over geometric cues to optimize for the 3D structure. Over the years, the incorporation of supervised learning allowed approaches to scale to more realistic settings and infer qualitative (e.g. <a href="http://dhoiem.cs.illinois.edu/projects/popup/">Hoiem et al.</a>) or quantitative (e.g. <a href="http://www.cs.cornell.edu/~asaxena/learningdepth/">Saxena et al.</a>) 3D representations. The trend of obtaining impressive results in realistic settings has since continued to the current CNN-based incarnations (e.g. <a href="http://www.cs.nyu.edu/~deigen/depth/">Eigen &amp; Fergus</a>, <a href="http://www.cs.cmu.edu/~xiaolonw/deep3d.html">Wang et al.</a>), but at the cost of increasing reliance on direct 3D supervision, making this paradigm rather restrictive. It is costly and painstaking, if not impossible, to obtain such supervision at a large scale. Instead, akin to the human visual system, we want our computational systems to <strong>learn 3D prediction without requiring 3D supervision</strong>.</p>

<p>With this goal in mind, our work and <a href="https://bair.berkeley.edu/blog/2017/07/11/confluence-of-geometry-and-learning/#refs">several other recent approaches</a> explore another form of supervision: multi-view observations, for learning single-view 3D. Interestingly, not only do these different works share the goal of incorporating multi-view supervision, the methodologies used also follow common principles. A unifying foundation to these approaches is the interaction between learning and geometry, where predictions made by the learning system are encouraged to be ‘geometrically consistent’ with the multi-view observations. Therefore, geometry acts as a bridge between the learning system and the multi-view training data.</p>

<h2 id="learning-via-geometric-consistency">Learning via Geometric Consistency</h2>
<p>Our aim is to to learn a <em>Predictor</em> $P$ (typically a neural network) that can infer 3D from a single 2D image. Under the supervision setting considered, the training data consists of multiple observations from different viewpoints. As alluded to earlier, geometry acts as a bridge to allow learning the <em>Predictor</em> $P$ using the training data. This is because we know precisely, in the form of concise geometric equations, the relationship between a 3D representation and the corresponding 2D projections.  We can therefore train $P$ to predict 3D that is <em>geometrically consistent</em> with the associated 2D observations.</p>

<p>
<img src="http://bair.berkeley.edu/static/blog/confluence/st1Ia9i.png" alt="geometric consistency">
</p>

<p>To illustrate the training process, consider a simple game between the <em>Predictor</em> $P$ and a geometry expert, the <em>Verifier</em> $V$. We give $P$ a single image $I$, and it predicts a 3D shape $S$. $V$, who is then given the prediction $S$, and an observation $O$ of the world from a different camera viewpoint $C$, uses the geometric equations to validate if these are consistent. We ask $P$ to predict $S$ that would pass this consistency check performed by $V$. The key insight is that since $P$ does not know $(O, C)$ which will be used to verify its prediction, it will have to predict $S$ that is consistent will <em>all</em> the possible observations (similar to the unknown ground-truth $S_{gt}$). This allows us to define the following training algorithm to learn 3D-from-2D prediction using only multi-view supervision.</p>

<ul>
  <li>Pick a random training image $I$ with associated observation $O$ from viewpoint $C$.</li>
  <li>Predict $S = P(I)$. Use $V$ to check consistency between $(S, O, C)$</li>
  <li>Update $P$, using gradient descent, to make $S$ more consistent with $(O, C)$.</li>
  <li>Repeat until convergence.</li>
</ul>

<p>The recent approaches pursuing single-view prediction using multi-view supervision all adhere to this template, the differences being the form of 3D prediction being pursued (e.g. depth or shape), and the kinds of multi-view observations needed (e.g. color images or foreground masks). We now look at two papers which push the boundaries of the multi-view supervision paradigm. The first one leverages classical ray consistency formulations to introduce a generic <em>Verifier</em> which can measure consistency between a 3D shape and diverse kinds of observations $O$. The second one demonstrates that it is possible to even further relax the supervision required and presents a technique to learn 3D-from-2D without even requiring the camera viewpoints $C$ for training.</p>

<h2 id="differentiable-ray-consistency">Differentiable Ray Consistency</h2>

<p>In our <a href="https://arxiv.org/pdf/1704.06254.pdf">recent paper</a>, we formulate a <em>Verifier</em> $V$ to measure the consistency between a 3D shape (represented as a probabilistic occupancy grid) and a 2D observation. Our generic formulation allows learning volumetric 3D prediction by leveraging different types of multi-view observations e.g. foreground masks, depth, color images, semantics etc. as supervision.</p>

<p>An insight which allows defining $V$ is that each pixel in the observation $O$ corresponds to a ray with some associated information. Then, instead of computing the geometric consistency between the observation $O$ and the shape $S$, we can consider, one ray at a time, the consistency between the shape $S$ and a ray $r$.</p>

<p>
<img src="http://bair.berkeley.edu/static/blog/confluence/NQaMdTl.png" alt="rays">
</p>

<p>The figure above depicts the various aspects of formulating the ray consistency cost. a) The predicted 3D shape and a sample ray with which we measure consistency. b,c) We trace the ray through the 3D shape and compute <em>event probabilities</em> - the probabilities that the ray terminates at various points on its path. d) We can measure how inconsistent each ray termination event is with the information available for that ray. e) By defining the ray consistency cost as the expected event cost, we can compute gradients for how the prediction should be updated to increase the consistency. While in this example we visualize a depth observation $O$, an advantage of our formulation is that it allows incorporating diverse kinds of observations (color images, foreground masks etc.) by simply defining the corresponding event cost function.</p>

<p>The results of 3D-from-2D prediction learned using our framework in different settings are shown below. Note that all the visualized predictions are obtained from a single RGB image by a <em>Predictor</em> $P$ trained <em>without using 3D supervision</em>.</p>

<table>
  <thead>
    <tr>
      <th><img src="http://bair.berkeley.edu/static/blog/confluence/sNetVis.png"></th>
      <th><img src="http://bair.berkeley.edu/static/blog/confluence/pascalVis.png"></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><sub><em>Results on ShapeNet dataset using multiple depth images as supervision for training. a) Input image. b,c) Predicted 3D shape.</em></sub></td>
      <td><sub><em>Results on PASCAL VOC dataset using pose and foreground masks as supervision for training. a) Input image. b,c) Predicted 3D shape.</em></sub></td>
    </tr>
  </tbody>
</table>

<table>
  <thead>
    <tr>
      <th><img src="http://bair.berkeley.edu/static/blog/confluence/csVis.png"></th>
      <th><img src="http://bair.berkeley.edu/static/blog/confluence/sNetColorVis.png"></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><sub><em>Results on Cityscapes dataset using  depth, semantics as supervision. a) Input image. b,c) Predicted 3D shape rendered under simulated forward motion.</em></sub></td>
      <td><sub><em>Results on ShapeNet dataset using multiple color images as supervision for training shape and per-voxel color prediction. a) Input image. b,c) Predicted 3D shape.</em></sub></td>
    </tr>
  </tbody>
</table>

<h2 id="learning-depth-and-pose-from-unlabeled-videos">Learning Depth and Pose from Unlabeled Videos</h2>
<p>Notice that in the above work, the input to the <em>Verifier</em> $V$ is an observation with <em>known</em> camera viewpoint/pose. This is reasonable from the perspective of an agent with sensorimotor functionality (e.g. human or robots with odometers), but prevents its applications to more unstructured data sources (e.g. videos). In another <a href="https://arxiv.org/abs/1704.07813">recent work</a>, we show that the pose requirement can be relaxed, and in fact jointly learned with the single image 3D predictor $P$.</p>

<p>
<img src="http://bair.berkeley.edu/static/blog/confluence/teaser_h.jpg" alt="problem setup">
</p>

<p>More specifically, our <em>Verifier</em> $V$ in this case is based on a <em>differentiable depth-based view synthesizer</em> that outputs a target view of the scene using the predicted depth map and pixels from a source view (i.e. observation) seen under a different camera pose. Here both the depth map and the camera pose are predicted, and the consistency is defined by the pixel reconstruction error between the synthesized and the ground-truth target view. By jointly learning the scene geometry and the camera pose, we are able to train the system on unlabeled video clips without any direct supervision for either depth or pose.</p>

<table>
  <thead>
    <tr>
      <th><img src="http://bair.berkeley.edu/static/blog/confluence/pipeline.jpg" alt="Training pipeline"></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><sub><em>Formulating the Verifier as a depth-based view synthesizer and joint learning of depth and camera pose allows us to train the entire system from unlabeled videos without any direct supervision for either depth or pose.</em> </sub></td>
    </tr>
  </tbody>
</table>

<p>We train and evaluate our model on the KITTI and Cityscapes datasets, which consist of videos captured by a car driving in urban cities. The video below shows frame-by-frame (i.e. no temporal smoothness) prediction made by our single-view depth network (more can be found in the <a href="https://people.eecs.berkeley.edu/~tinghuiz/projects/SfMLearner/">project webage</a>).</p>

<p>
<img src="http://bair.berkeley.edu/static/blog/confluence/cityscapes_sample_results.gif?raw=true" alt="depth_gif">
</p>

<p>Surprisingly, despite being trained without any ground-truth labels, our single-view depth model performs on par with some of the supervised baselines, while the pose estimation model is also comparable with well-established SLAM systems (see the <a href="https://arxiv.org/pdf/1704.07813.pdf">paper</a> for more details).</p>

<hr>

<p>Learning single image 3D without 3D supervision is an exciting and thriving topic in computer vision. Using geometry as a bridge between the learning system and the multi-view training data allows us to bypass the tedious and expensive process of acquiring ground-truth 3D labels. More broadly, one could interpret the geometric consistency as a form of <em>meta supervision</em> on not <em>what</em> the prediction is but <em>how</em> it should behave. We believe that similar principles could be applied to other problem domains where obtaining direct labels is difficult or infeasible.</p>

<hr>

<p><em>We would like to thank <a href="https://people.eecs.berkeley.edu/~efros/">TZ’s advisor</a> and <a href="https://people.eecs.berkeley.edu/~malik/">TZ’s advisor’s advisor</a> for their valuable feedback.</em></p>

<p><strong>This post is based on the following papers:</strong></p>
<ul>
  <li>
    <p><a href="https://shubhtuls.github.io/drc/"><em>Multi-view Supervision for Single-view Reconstruction via Differentiable Ray Consistency.</em></a>  <br>
S. Tulsiani, T. Zhou, A. A. Efros, J. Malik. In CVPR, 2017 . (<a href="https://arxiv.org/pdf/1704.06254.pdf">pdf</a>, <a href="https://github.com/shubhtuls/drc">code</a>, <a href="https://shubhtuls.github.io/drc/">webpage</a>)</p>
  </li>
  <li>
    <p><a href="https://people.eecs.berkeley.edu/~tinghuiz/projects/SfMLearner/"><em>Unsupervised Learning of Depth and Ego-Motion from Video.</em></a> <br>
T. Zhou, M. Brown, N. Snavely, D. Lowe. In CVPR, 2017. (<a href="https://arxiv.org/pdf/1704.07813.pdf">pdf</a>, <a href="https://github.com/tinghuiz/SfMLearner">code</a>, <a href="https://people.eecs.berkeley.edu/~tinghuiz/projects/SfMLearner/">webpage</a>)</p>
  </li>
</ul>

<p><strong>Other recent multi-view supervised 3D prediction approaches:</strong><a name="refs"></a></p>

<ul>
  <li>
    <p><a href="https://arxiv.org/abs/1603.04992"><em>Unsupervised CNN for Single View Depth Estimation: Geometry to the Rescue.</em></a> <br>
R. Garg, B. G. Vijay Kumar, G. Carneiro, I. Reid. In ECCV, 2016.</p>
  </li>
  <li>
    <p><a href="https://sites.google.com/site/skywalkeryxc/perspective_transformer_nets"><em>Perspective Transformer Nets: Learning Single-View 3D Object Reconstruction without 3D Supervision.</em></a>  <br>
X. Yan, J. Yang, E. Yumer, Y. Guo, H. Lee. In NIPS, 2016.</p>
  </li>
  <li>
    <p><a href="https://arxiv.org/abs/1607.00662"><em>Unsupervised Learning of 3D Structure from Images.</em></a>  <br>
D. J. Rezende, S. M. Ali Eslami, S. Mohamed, P. Battaglia, M. Jaderberg, N. Heess. In NIPS, 2016.</p>
  </li>
  <li>
    <p><a href="http://mgadelha.me/home/prgan/index.html"><em>3D Shape Induction from 2D Views of Multiple Objects.</em></a> <br>
M. Gadelha, S. Maji, R. Wang. arXiv preprint, 2016.</p>
  </li>
  <li>
    <p><a href="http://visual.cs.ucl.ac.uk/pubs/monoDepth/"><em>Unsupervised Monocular Depth Estimation with Left-Right Consistency.</em></a> <br>
C. Godard, O. M. Aodha, G. J. Brostow. In CVPR, 2017.</p>
  </li>
</ul>

</article>

</div><br>
<br>
<br>
<br>
<br>
<br>
via The Berkeley Artificial Intelligence Research Blog https://ift.tt/2EPVVfk<br>
<br>
October 24, 2018 at 02:22PM
